hydrolib/core/dflowfm/research/models.py

```diff
@@ -0,0 +1,903 @@
+    )
+    research_metadatafile: Optional[DiskOnlyFileModel] = Field(
+        None, alias="metadatafile"
+    )
+    research_mbainterval: Optional[float] = Field(None, alias="mbainterval")
+    research_velocitydirectionclassesinterval: Optional[int] = Field(
+        None, alias="velocitydirectionclassesinterval"
+    )
+    research_wrirst_bnd: Optional[bool] = Field(None, alias="wrirst_bnd")
+    research_generateuuid: Optional[bool] = Field(None, alias="generateuuid")
+    research_timesplitinterval: Optional[str] = Field(None, alias="timesplitinterval")
+    research_rugfile: Optional[PolyFile] = Field(None, alias="rugfile")
+    research_mbawritecsv: Optional[bool] = Field(None, alias="mbawritecsv")
+    research_mbalumpfromtomba: Optional[bool] = Field(None, alias="mbalumpfromtomba")
+    research_mbalumpprocesses: Optional[bool] = Field(None, alias="mbalumpprocesses")
+    research_waqvertaggr: Optional[DiskOnlyFileModel] = Field(None, alias="waqvertaggr")
+    research_mbawritenetcdf: Optional[bool] = Field(None, alias="mbawritenetcdf")
+    research_mbawritetxt: Optional[bool] = Field(None, alias="mbawritetxt")
+    research_nccompression: Optional[bool] = Field(None, alias="nccompression")
+    research_wrimap_ice: Optional[bool] = Field(None, alias="wrimap_ice")
+    research_wrimap_trachytopes: Optional[bool] = Field(
+        None, alias="wrimap_trachytopes"
+    )
+
+    _split_to_list = get_split_string_on_delimiter_validator(
+        "research_velocitymagnitudeclasses",
+    )
+
+
+class ResearchProcesses(Processes):
+    """An extended [processes] section that includes highly experimental research keywords."""
+
+    class Comments(Processes.Comments):
+        substancedensitycoupling: Optional[str] = Field(
+            "Substance rho coupling (0=no, 1=yes).", alias="substancedensitycoupling"
+        )
+
+    comments: Comments = Comments()
+
+    research_substancedensitycoupling: Optional[bool] = Field(
+        None, alias="substancedensitycoupling"
+    )
+
+
+class ResearchFMModel(FMModel):
+    """
+    An extended FMModel that includes highly experimental research sections and keywords.
+    """
+
+    general: ResearchGeneral = Field(default_factory=ResearchGeneral)
+    geometry: ResearchGeometry = Field(default_factory=ResearchGeometry)
+    numerics: ResearchNumerics = Field(default_factory=ResearchNumerics)
+    physics: ResearchPhysics = Field(default_factory=ResearchPhysics)
+    sediment: ResearchSediment = Field(default_factory=ResearchSediment)
+    wind: ResearchWind = Field(default_factory=ResearchWind)
+    waves: ResearchWaves = Field(default_factory=ResearchWaves)
+    time: ResearchTime = Field(default_factory=ResearchTime)
+    restart: ResearchRestart = Field(default_factory=ResearchRestart)
+    trachytopes: ResearchTrachytopes = Field(default_factory=ResearchTrachytopes)
+    output: ResearchOutput = Field(default_factory=ResearchOutput)
+    processes: Optional[ResearchProcesses] = Field(None)
```

Comment:
is this Field(None) correct?

Reply:
Yes. The [processes] section is Optional.